lib/formats/bs_format_ripejson.c

```diff
@@ -0,0 +1,669 @@
+  if ((hexstr_len & 0x1) != 0 || hexstr_len > (4096*2)){
+    return -1;
+  }
+  uint16_t msg_len = (hexstr_len / 2) + 2 + 1;
+  uint16_t tmp = htons(msg_len);
+  uint8_t *bufp = buf;
+
+  if (msg_len > buflen) {
+    return -1;
+  }
+
+  // populate the message header (but don't include the marker)
+  memcpy(buf, &tmp, sizeof(tmp));
+  buf[2] = msg_type;
+
+  // parse the hex string, one nybble at a time
+  bufp = buf + 3;
+  hexstr_to_bytes(bufp, hexstr, hexstr_len);
+
+  return msg_len;
+}
+
+/* ====================================================================== */
+/* ====================================================================== */
+/* ==================== PRIVATE FUNCTIONS BELOW HERE ==================== */
+/* ====================================================================== */
+/* ====================================================================== */
+
+void process_common_fields(bgpstream_format_t *format, bgpstream_record_t *record){
+
+  // populate collector name
+  memcpy(record -> collector_name , STATE->json_fields.host.ptr, STATE->json_fields.host.len);
+  record -> collector_name[STATE->json_fields.host.len] = '\0';
+
+  // populate peer asn
+  RDATA->elem->peer_asn = (uint32_t) strtol((char *)STATE->json_fields.peer_asn.ptr, NULL, 10);
+
+  // populate peer ip
+  bgpstream_addr_storage_t addr;
+  memcpy(STATE->field_buffer, STATE->json_fields.peer.ptr, STATE->json_fields.peer.len);
+  STATE->field_buffer[STATE->json_fields.peer.len] = '\0';
+  bgpstream_str2addr((char *)STATE->field_buffer, &addr);
+  bgpstream_addr_copy((bgpstream_ip_addr_t *)&RDATA->elem->peer_ip,
+                      (bgpstream_ip_addr_t *)&addr);
+  // populate time-stamp
+  double time_double = strtod((char *)STATE->json_fields.timestamp.ptr, NULL);
+  record->time_sec = (int) time_double;
+  record->time_usec = (int)((time_double - (int) time_double) * 1000000);
+
+}
+
+int process_update_message(bgpstream_format_t *format, bgpstream_record_t *record){
+
+  // convert body to bytes
+  size_t dec_len =  (size_t) hexstr_to_bgpmsg(STATE->json_bytes_buffer,
+                                      4096,
+                                      (char*) STATE->json_fields.body.ptr,
+                                      STATE->json_fields.body.len,
+                                      PARSEBGP_BGP_TYPE_UPDATE);
+  size_t err;
+
+  RDATA->msg->type = PARSEBGP_MSG_TYPE_BGP;
+  if ((err = parsebgp_decode(STATE->opts, RDATA->msg->type, RDATA->msg,
+                             STATE->json_bytes_buffer, &dec_len)) != PARSEBGP_OK) {
+    fprintf(stderr, "ERROR: Failed to parse message (%zu:%s)\n", err, parsebgp_strerror(err));
+    parsebgp_clear_msg(RDATA->msg);
+    return -1;
+  }
+
+  process_common_fields(format, record);
+
+  return 0;
+}
+
+int process_status_message(bgpstream_format_t *format, bgpstream_record_t *record){
+
+  // process direction
+  if( bcmp(&"down", STATE->json_fields.state.ptr, STATE->json_fields.state.len) == 0){
```

Comment:
you shouldn't need to take the address of the "down" string.
you are also guaranteed to do something bad when the length of the json string is longer than the string "down"
you need to pass the shortest of the two lengths. but a better approach is to first check that the lengths are identical before calling bcmp..